<launch>

    <arg name="instances" default="1" />

    <group ns="$(eval 'node' + str(arg('instances')))">

        <node name="prediction_server" pkg="perception_pipeline" type="prediction_service.py" output="screen">

            <param name="instance_id" type="int" value="$(eval arg('instances'))" />
        </node>

    </group>

    <include file="$(find perception_pipeline)/launch/prediction_services.launch" if="$(eval arg('instances') - 1 > 0)">

        <arg name="instances" value="$(eval arg('instances')-1)" />

    </include>

    <!-- 
    <node name="prediction_server_1" pkg="perception_pipeline" type="prediction_service.py" output="screen">
        <param name="instance_id" type="int" value="1" />
    </node>

    <node name="prediction_server_2" pkg="perception_pipeline" type="prediction_service.py" output="screen">
        <param name="instance_id" type="int" value="2" />
    </node>

    <node name="prediction_server_3" pkg="perception_pipeline" type="prediction_service.py" output="screen">
        <param name="instance_id" type="int" value="3" />
    </node>

    <node name="prediction_server_4" pkg="perception_pipeline" type="prediction_service.py" output="screen">
        <param name="instance_id" type="int" value="4" />
    </node>
     -->

</launch>
